rust/chains/hyperlane-cosmos/src/providers/mod.rs

```diff
@@ -172,14 +204,22 @@ impl HyperlaneProvider for CosmosProvider {
             .map_err(|e| ChainCommunicationError::from_other_str("could not convert from proto"))?;
         let contract = H256::try_from(CosmosAccountId::new(&msg.contract))?;
 
-        // TODO choose correct signer (should it be the one paid for the transaction)
-        let signer = tx
+        let (sender, nonce) = tx
```

Comment:
a comment about the general approach here may be useful for future readers. Iiuc it'll try using the fee payer if it exists / is able to get a signer there, otherwise it falls back to getting the first signer?

Reply:
It makes sense. I added a comment.